### PR TITLE
[Refactor] Convert LocalizationProvider to protocol

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Convert LocalizationProvider to protocol
* Added LocalizationProvider to dependency injection for CallingViewModel and SetupViewModel (top level only)
* Updates unit test SUT for CallingViewModel and SetupViewModel
* Added unit test for LocalizationProvider

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ x ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Ensure test passes, and app still functional

## Other Information

* LocalizationProvider isRightToLeft test will be added in separate PR

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift [hunk 1]
```diff
@@ -140,9 +140,8 @@ public class CallComposite {
         }
     }
 
-    private func setupLocalization() {
+    private func setupLocalization(localizationProvider: LocalizationProvider) {
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift:143]: nit suggestion
```suggestion
    private func setupLocalization(with provider: LocalizationProvider) {
```

File: AzureCommunicationUI/AzureCommunicationUITests/LocalizationProviderMocking.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    func applyLocalizationConfiguration(_ localeConfig: LocalizationConfiguration) {
+        return
+    }
+
+    func getSupportedLanguages() -> [String] {
+        return ["en"]
+    }
+
+    func getLocalizedString(_ key: String) -> String {
+        return key
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUITests/LocalizationProviderMocking.swift:20]: I thought we were going to return a constant string here like "ABC"

File: AzureCommunicationUI/AzureCommunicationUITests/LocalizationProviderMocking.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    func getLocalizedString(_ key: String, _ args: CVarArg...) -> String {
+        return key
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUITests/LocalizationProviderMocking.swift:24]: Same comment as above here in this method

**Review by JoshuaLai — APPROVED** [review]
Nothing to super pressing.  Mainly a few questions and one nitpick suggestion. 
The changes look good and tested it is working as expected.

**Comment by palatter** [comment]
Just a minor comment but it looks good and tested!

**Review by palatter — APPROVED** [review]

**Pull request merged**